src/fetch/period_slippage.py

```diff
@@ -27,22 +30,43 @@ def prepend_to_sub_query(query: str, table_to_add: str) -> str:
 
 def add_token_list_table_to_query(original_sub_query: str) -> str:
     """Inserts the token_list table right after the WITH statement into the sql query"""
-    token_list = get_trusted_tokens_from_url(ALLOWED_TOKEN_LIST_URL)
-    sql_query_for_allowed_token_list = allowed_token_list_query(token_list)
-    return prepend_to_sub_query(original_sub_query, sql_query_for_allowed_token_list)
+    token_list = fetch_trusted_tokens()
+    allowed_tokens_query = allowed_token_list_query(token_list)
+    return prepend_to_sub_query(original_sub_query, allowed_tokens_query)
 
 
-def slippage_query(dune: DuneAnalytics) -> str:
-    """Constructs our slippage query by joining sub-queries"""
-    path = "./queries/slippage"
-    slippage_sub_query = dune.open_query(
-        File("subquery_batchwise_internal_transfers.sql", path).filename()
-    )
-    select_slippage_query = dune.open_query(
-        File("select_slippage_results.sql", path).filename()
-    )
+class QueryType(Enum):
+    """
+    Determines type of slippage data to be fetched.
+    The slippage subquery allows us to select from either of the two result tables defined here.
+    """
+
+    PER_TX = "results_per_tx"
+    TOTAL = "results"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+def slippage_query(query_type: QueryType = QueryType.TOTAL) -> str:
```

Comment:
Except that Per TX is only ever called in unit testing.

Reply:
This refactoring is only a segway into what I actually envision for this project. 

Currently the slippage query is handling both AMM slippage and the internal balance penalty. Realistically, we should be fetching the classified internal token imbalances, classifying those which are due to slippage and those which are internal in the python code. A reason for this is to be able to log the slippage penalty separately from the internal balance and illegal trading penalty.

Should really open an issue/discussion for this. 

In any case, the TOTAL is default because that is what the scripts make use of and the PER_TX is used internally for debugging and testing.